astronomer/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -44,12 +45,43 @@ async def get_client_async(self) -> AioBaseClient:
         )
 
         async_connection = get_session()
+        session_token = conn_config.aws_session_token
+        aws_secret = conn_config.aws_secret_access_key
+        aws_access = conn_config.aws_access_key_id
+        if conn_config.role_arn:
+            credentials = await self.get_role_credentials(
+                async_session=async_connection, conn_config=conn_config
+            )
+            if credentials:
+                session_token = credentials["SessionToken"]
+                aws_access = credentials["AccessKeyId"]
+                aws_secret = credentials["SecretAccessKey"]
         return async_connection.create_client(
             service_name=self.client_type,
             region_name=conn_config.region_name,
-            aws_secret_access_key=conn_config.aws_secret_access_key,
-            aws_access_key_id=conn_config.aws_access_key_id,
-            aws_session_token=conn_config.aws_session_token,
+            aws_secret_access_key=aws_secret,
+            aws_access_key_id=aws_access,
+            aws_session_token=session_token,
             verify=self.verify,
             config=self.config,
+            endpoint_url=conn_config.endpoint_url,
         )
+
+    @staticmethod
+    async def get_role_credentials(
+        async_session: AioSession, conn_config: AwsConnectionWrapper
+    ) -> Optional[Dict[str, str]]:
+        """Get the role_arn, method credentials from connection details and get the role credentials detail"""
+        async with async_session.create_client(
+            "sts",
+            aws_access_key_id=conn_config.aws_access_key_id,
+            aws_secret_access_key=conn_config.aws_secret_access_key,
+        ) as client:
+            if conn_config.assume_role_method == "assume_role" or conn_config.assume_role_method is None:
+                response: Dict[str, Dict[str, str]] = await client.assume_role(
+                    RoleArn=conn_config.role_arn,
+                    RoleSessionName="RoleSession",
+                    **conn_config.assume_role_kwargs,
+                )
+                return response["Credentials"]
+            return None  # pragma: no cover
```

Comment:
why no cover, what about just removing this line?

Reply:
+1

Reply:
Initially, there wasn't a `return None`, Mypy was complaining the _function is expecting return_ so I have added this and code coverage was complaining, so I have added this